utils/src/server/helpers.rs

```diff
@@ -44,8 +44,8 @@ pub async fn collect_batch_response(rx: mpsc::UnboundedReceiver<String>) -> Stri
 	let mut buf = String::with_capacity(2048);
 	buf.push('[');
 	let mut buf = rx
-		.fold(buf, |mut acc, response| async {
-			acc = [acc, response].concat();
```

Comment:
I assumed this would alloc since `Concat` is implemented on `Borrow<str>`, and just checking ASM produced for this, it does indeed alloc on every call here, so the capacity isn't even being used.